sky/skylet/providers/command_runner.py

```diff
@@ -0,0 +1,258 @@
+"""Sky's DockerCommandRunner."""
+import json
+import os
+from typing import Dict
+
+from sky.skylet import constants
+
+from ray.autoscaler._private.cli_logger import cli_logger
+from ray.autoscaler._private.command_runner import DockerCommandRunner
+from ray.autoscaler._private.docker import check_docker_running_cmd
+
+
+def docker_start_cmds(
+    user,
+    image,
+    mount_dict,
+    container_name,
+    user_options,
+    cluster_name,
+    home_directory,
+    docker_cmd,
+):
+    """Generating docker start command without --rm.
+    
+    The code is borrowed from `ray.autoscaler._private.docker`. The only
+    difference is that we don't use `--rm` flag here.
+    """
+    del user  # unused
+
+    # pylint: disable=import-outside-toplevel
+    from ray.autoscaler.sdk import get_docker_host_mount_location
+
+    docker_mount_prefix = get_docker_host_mount_location(cluster_name)
+    mount = {f'{docker_mount_prefix}/{dst}': dst for dst in mount_dict}
+
+    mount_flags = ' '.join([
+        '-v {src}:{dest}'.format(src=k,
+                                 dest=v.replace('~/', home_directory + '/'))
+        for k, v in mount.items()
+    ])
+
+    # for click, used in ray cli
+    env_vars = {'LC_ALL': 'C.UTF-8', 'LANG': 'C.UTF-8'}
+    env_flags = ' '.join(
+        ['-e {name}={val}'.format(name=k, val=v) for k, v in env_vars.items()])
+
+    user_options_str = ' '.join(user_options)
+    docker_run = [
+        docker_cmd,
+        'run',
+        '--name {}'.format(container_name),
+        '-d',
+        '-it',
+        mount_flags,
+        env_flags,
+        user_options_str,
+        '--net=host',
+        image,
+        'bash',
+    ]
+    return ' '.join(docker_run)
+
+
+class SkyDockerCommandRunner(DockerCommandRunner):
```

Comment:
Can we annotate the palce where we update against original `DockerCommandRunner` with comments starting with `skypilot:`? Reason: in the future if we are going to update the command runner from the upstream, it will be easier to do that.

Reply:
Added now